src/matchbox/core.cljc

```diff
@@ -506,6 +506,12 @@
 
 ;; onAuth and offAuth are not wrapped yet
 
+(defn onAuth [ref cb]
+  (.onAuth ref (wrap-auth-cb cb)))
+
+(defn offAuth [ref cb]
+  (.offAuth ref cb))
```

Comment:
Ah I see - there's still an issue here though. Since we're wrapping the original callback, we need that decorated copy to use for deregistering. And I'm guessing `wrap-auth-cb` breaks reference equality, so we can't just wrap again. We'll probably need to store some state somewhere 😞 

Also need to consider the JVM case (`addAuthStateListener` and `removeAuthStateListener` I believe)